### PR TITLE
Add parser for CAids

### PR DESCRIPTION
**Pull request by helensch**
For ENSVAR-1157.
Only for input parsing, not for output
Dependent on fetch_all_by_caid in https://github.com/Ensembl/ensembl-variation/pull/689

### CODE REVIEW

File: modules/Bio/EnsEMBL/VEP/Parser/CAID.pm [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+=cut
+
+sub new {
+  my $caller = shift;
+  my $class = ref($caller) || $caller;
+
+  my $self = $class->SUPER::new(@_);
+
+  # requires db connection
+  throw("ERROR: Cannot use CAID format in offline mode") if $self->param('offline');
```
Comment (dglemos) [inline modules/Bio/EnsEMBL/VEP/Parser/CAID.pm:103]: Could change to CAid.

File: modules/Bio/EnsEMBL/VEP/Parser/CAID.pm [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+
+=head2 create_VariationFeatures
+
+  Example    : $vfs = $parser->create_VariationFeatures();
+  Description: Create one or more VariationFeature objects from the current line
+               of input; multiple may be returned if a CAid maps to multiple
+               variant alleles
+  Returntype : arrayref of Bio::EnsEMBL::VariationFeature
+  Exceptions : warns if CAID not found in database
```
Comment (dglemos) [inline modules/Bio/EnsEMBL/VEP/Parser/CAID.pm:135]: Same here, CAid.

**Review by dglemos — COMMENTED** [review]
I tested VEP and Variant Recoder with different input ids and options. Looks good. 
After https://github.com/Ensembl/ensembl-variation/pull/689 has been merged, all tests pass.

File: modules/Bio/EnsEMBL/VEP/VariantRecoder.pm [hunk 3]
```diff
@@ -106,7 +106,7 @@ sub new {
     buffer_size
   );
 
-  $config->{fields} ||= 'id,hgvsg,hgvsc,hgvsp,spdi';
+  $config->{fields} ||= 'id,hgvsg,hgvsc,hgvsp,spdi,caid';
```
[the remark this answers is not in the document]
Comment (helensch) [reply modules/Bio/EnsEMBL/VEP/VariantRecoder.pm]: caid has been removed

File: modules/Bio/EnsEMBL/VEP/Parser/CAID.pm (hunk 1 above)
Comment (helensch) [reply modules/Bio/EnsEMBL/VEP/Parser/CAID.pm]: Changed to CAid

File: modules/Bio/EnsEMBL/VEP/Parser/CAID.pm (hunk 2 above)
Comment (helensch) [reply modules/Bio/EnsEMBL/VEP/Parser/CAID.pm]: Changed to CAid

**Comment by helensch** [comment]
@dglemos thank you for review. I have made changes requested.

**Pull request merged**